pkg/local_object_storage/engine/shards.go

```diff
@@ -87,6 +87,9 @@ func (e *StorageEngine) createShard(opts []shard.Option) (*shard.Shard, error) {
 		shard.WithExpiredTombstonesCallback(e.processExpiredTombstones),
 		shard.WithExpiredLocksCallback(e.processExpiredLocks),
 		shard.WithDeletedLockCallback(e.processDeletedLocks),
+		shard.WithReportErrorFunc(func(msg string, err error) {
+			e.reportShardErrorBackground(id.String(), msg, err)
```

Comment:
Could be a bug here like with metrics: `id` is generated, but could be updated below.